### ISSUE TITLE
flake8 should only consider a noqa comment if it is actually a comment

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#1321]
minimal reproduction:

```python
def f():
    x = '# noqa'
```

I expect this to report the same as if that were an unrelated string:

```python
def f():
    x = '......'
```

```console
$ flake8 unrelated.py
unrelated.py:2:5: F841 local variable 'x' is assigned to but never used
```

**Comment by sco1** [comment]
Did you already have something in mind to fix this?

If I'm following correctly (disclaimer: I'm new to Flake8's internals), the inline `noqa` check comes from [this regex](https://github.com/PyCQA/flake8/blob/84c95766e679710d9a0fa910a1cd276c87be42c5/src/flake8/defaults.py#L27-L42), is it desired to come up with a solution to this that fits into the regex? Or, maybe overthinking, can we do something like tokenize the line if the regex matches?

**Comment by asottile** [comment]
yeah I'm planning to maybe rework this into a pipeline based on the tokenization (since that has to happen for linting anyway) -- I don't really want to shoehorn it into the existing structure as I'll probably be reworking that.  I think the regex is still relevant but it should only be tested against `COMMENT` tokens
